publications app

Theoabw opened the pull request.

Theoabw committed:
initial publications app

Theoabw committed:
initial publications app

theoabw commented:
TODO: 
1. Styling on mobile and overall
2. Custom slug functionality
3. Make sure it works with S3

jaspnas commented on static/common/publications/pdf.mjs:
Why are we not downloading these from a CDN?

jaspnas commented on static/common/publications/pdf.worker.mjs:
Why are we not downloading these from a CDN?

Theoabw committed:
fixed js and style a bit

Theoabw committed:
fixed js and style a bit

jaspnas commented on publications/views.py:
```diff
@@ -0,0 +1,20 @@
+    try:
+        pdf_url = pdf_file.get_file_url()
+    except ClientError as e:
+        return JsonResponse({'error': str(e)}, status=404)
```
Let's not return JsonResponse, but let django handle it to show the default 404 page

jaspnas commented on publications/models.py:
```diff
@@ -0,0 +1,27 @@
+
+class PDFFile(models.Model):
+    title = models.CharField(_('Title'), max_length=250)
+    file = PrivateFileField(upload_to=upload_to)
```
We probably don't want to store these files in private storage as publications are intended to be publicly viewable.

jaspnas commented on publications/models.py:
```diff
@@ -0,0 +1,27 @@
+    return f'pdfs/{filename}'
+
+
+class PDFFile(models.Model):
```
We want to add slug support

Theoabw committed:
revamp + slugs

Theoabw committed:
revamp + slugs

Theoabw committed:
view changes

Theoabw committed:
view changes

Theoabw committed:
add migration

Theoabw committed:
add migration

Theoabw committed:
style improvements

Theoabw committed:
style improvements

jaspnas commented on core/settings/common.py:
```diff
@@ -233,6 +233,7 @@ def get_staff_groups(default_groups: list):
 else:
     MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
     MEDIA_URL = '/media/'
+    DEFAULT_FILE_STORAGE = 'django.core.files.storage.FileSystemStorage'
```
This is deprecated and removed in django 5.1

jaspnas commented on publications/models.py:
```diff
@@ -0,0 +1,64 @@
+    title = models.CharField(_('Title'), max_length=250)
+    slug = models.SlugField(_('Slug'), max_length=255, unique=True, blank=True,
+                            help_text=_('Leave empty to auto-generate from title'))
+    file = models.FileField(_('File'), upload_to=upload_to, storage=PublicMediaStorage() if settings.USE_S3 else None)
```
We shouldn't have if statements here, see how archive is doing this by redefining save()

jaspnas commented on publications/views.py:
```diff
@@ -0,0 +1,22 @@
+from .models import PDFFile
+from django.core.paginator import Paginator
+
+@login_required
```
I don't know if we want login required for all PDF's, I think UpDate is currently stored in a public static page

jaspnas commented on publications/models.py:
```diff
@@ -0,0 +1,64 @@
+
+        super().save(*args, **kwargs)
+
+    def count_pdf_pages(self):
```
Do we need to count the pages on the backend, or can we have the client do it?

theoabw replied:
We don't actually need this feature and it's not really critical for it to work so I'll probably just remove it for now. Future issue can add a client side counter if anyone wants to do that.

jaspnas commented on publications/views.py:
```diff
@@ -0,0 +1,22 @@
+    paginator = Paginator(pdfs, 10)  # Show 10 PDFs per page
+    page_number = request.GET.get('page')
+    page_obj = paginator.get_page(page_number)
+    return render(request, 'publications/list.html', {'page_obj': page_obj})
```
Template not found

Theoabw committed:
more style improvements & list

Theoabw committed:
more style improvements & list

Theoabw committed:
model and view improvements

Theoabw committed:
B
Merge branch 'feat/612-feat-publications-app' of https://github.com/Datateknologerna-vid-Abo-Akademi/date-website into feat/612-feat-publications-app

jaspnas commented on core/settings/common.py:
```diff
@@ -229,10 +229,35 @@ def get_staff_groups(default_groups: list):
 else:
     MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
     MEDIA_URL = '/media/'
+    DEFAULT_FILE_STORAGE = 'django.core.files.storage.FileSystemStorage'
```
Should probably remove this

jaspnas commented on requirements.txt:
```diff
@@ -26,3 +26,4 @@ django_cleanup
 daphne
 celery[redis]
 gunicorn
+pymupdf
```
This should be removed if we don't need it

jaspnas commented on publications/models.py:
```diff
@@ -0,0 +1,65 @@
+    author = models.CharField(_('Author'), max_length=250, blank=True)
+    publication_date = models.DateField(_('Publication Date'), null=True, blank=True)
+    file_size = models.PositiveIntegerField(_('File Size (bytes)'), editable=False, null=True)
+    is_public = models.BooleanField(_('Public Access'), default=False,
```
This should be true by default

jaspnas commented on publications/models.py:
```diff
@@ -0,0 +1,65 @@
+    updated_at = models.DateTimeField(_('Updated at'), auto_now=True)
+    author = models.CharField(_('Author'), max_length=250, blank=True)
+    publication_date = models.DateField(_('Publication Date'), null=True, blank=True)
+    file_size = models.PositiveIntegerField(_('File Size (bytes)'), editable=False, null=True)
```
Not needed

jaspnas commented on publications/models.py:
```diff
@@ -0,0 +1,65 @@
+    description = models.TextField(_('Description'), blank=True)
+    uploaded_at = models.DateTimeField(_('Uploaded at'), auto_now_add=True)
+    updated_at = models.DateTimeField(_('Updated at'), auto_now=True)
+    author = models.CharField(_('Author'), max_length=250, blank=True)
```
Is this needed?

jaspnas commented on publications/models.py:
```diff
@@ -0,0 +1,65 @@
+    uploaded_at = models.DateTimeField(_('Uploaded at'), auto_now_add=True)
+    updated_at = models.DateTimeField(_('Updated at'), auto_now=True)
+    author = models.CharField(_('Author'), max_length=250, blank=True)
+    publication_date = models.DateField(_('Publication Date'), null=True, blank=True)
```
Should be autofilled as now

jaspnas commented:
Color theme doesn't look quite right 
![image](https://github.com/user-attachments/assets/e99cad15-281e-412d-8821-2a340d77087d)

jaspnas commented on publications/admin.py:
```diff
@@ -0,0 +1,38 @@
+            'fields': ('title', 'slug', 'author', 'publication_date', 'description', 'file')
+        }),
+        ('Access Control', {
+            'fields': ('is_public', 'requires_login'),
```
These feel like duplicates

theoabw replied:
One is if anyone (including members) gets to see it and the other is for if you have to be logged in to see it

jaspnas commented on static/common/publications/js/pdfViewer.js:
```diff
@@ -0,0 +1,84 @@
+
+function updateLoadingProgress(percent) {
+    // Implement loading progress UI update
+    console.log(`Loading PDF: ${percent.toFixed(2)}%`);
```
This is spammy in the console

jaspnas reviewed (changes requested):
Seems to work on s3, shouldn't break anything existing. Let's make admin a bit clearer and then we can merge this.

Theoabw committed:
clean up & style improvements

theoabw commented:
> Color theme doesn't look quite right ![image](https://private-user-images.githubusercontent.com/46525212/360386034-e99cad15-281e-412d-8821-2a340d77087d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MTk0NDcsIm5iZiI6MTcyNDQxOTE0NywicGF0aCI6Ii80NjUyNTIxMi8zNjAzODYwMzQtZTk5Y2FkMTUtMjgxZS00MTJkLTg4MjEtMmEzNDBkNzcwODdkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDEzMTkwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlZGIzNzE2ZTIxNDBlZmM0ZmI1NTAzZjRlNDMyYjljY2VjNDllNTQ3Y2U4YTk2OTlhZWZkYzE5YWY2MjAzNTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.L_Lub-XbB4x9fsbmDbvHNC6dSOtzWyJacyvFRI8v7qE)

dunno what the colors should be. Feel free to change it if you want to

andreedd commented:
You could bring it to life a bit more with some icons.

It could kind of look like the events:
<img width="749" alt="Screenshot 2024-08-23 at 17 27 09" src="https://github.com/user-attachments/assets/b480dc2c-5d69-4dba-9acd-1a6fff252714">

Colors can be find in root.css.

jaspnas commented on core/settings/common.py:
```diff
@@ -249,14 +258,12 @@ def get_staff_groups(default_groups: list):
                 "location": PUBLIC_MEDIA_LOCATION,
             }
         },
-        "staticfiles": {
```
You can't partially redefine storages here if you define it on root level

jaspnas committed:
Fix publications
